org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintEclipseHeadlessRpcClient.java

```diff
@@ -59,6 +67,16 @@
  *
  */
 public abstract class SonarLintEclipseHeadlessRpcClient implements SonarLintRpcClientDelegate {
+  
+  @Override
+  public Path getBaseDir(String configurationScopeId) throws ConfigScopeNotFoundException {
+    var project = resolveProject(configurationScopeId);
+    var projectLocation = project.getResource().getLocation();
+    // In some unfrequent cases the project may be virtual and don't have physical location
+    // so fallback to use analysis work dir that was created before
+    var analysisWorkDir = project.getWorkingDir().resolve("sonarlint");
```

Comment:
This doesn't make any sense to me. Where is the magic `sonarlint` folder coming from? I've never encountered that and I also didn't find any clue where this folder would have been created.

Reply:
I just reused the same piece of code that we were using before, see https://github.com/SonarSource/sonarlint-eclipse/blob/10.0.1.81733/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AbstractAnalyzeProjectJob.java#L180

Reply:
And the continuation here: https://github.com/SonarSource/sonarlint-eclipse/blob/10.0.1.81733/org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AbstractAnalyzeProjectJob.java#L259